monix-tail/shared/src/main/scala/monix/tail/internal/IterantDropWhileWithIndex.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2014-2018 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.tail.internal
+
+import cats.effect.Sync
+import cats.syntax.all._
+import monix.execution.misc.NonFatal
+import monix.tail.Iterant
+import monix.tail.Iterant.{Halt, Last, Next, NextBatch, NextCursor, Suspend}
+import monix.tail.batches.BatchCursor
+import scala.annotation.tailrec
+
+private[tail] object IterantDropWhileWithIndex {
+  /**
+    * Implementation for `Iterant#dropWhileWithIndex`
+    */
+  def apply[F[_], A](source: Iterant[F, A], p: (A, Int) => Boolean)
+                    (implicit F: Sync[F]): Iterant[F, A] = {
+
+    // Reusable logic for NextCursor / NextBatch branches
+    @tailrec
+    def evalCursor(ref: F[Iterant[F, A]], cursor: BatchCursor[A], rest: F[Iterant[F, A]],
+                   stop: F[Unit], dropped: Int, index: Int, continueDropping: Boolean): Iterant[F, A] = {
```

Comment:
Personal pet peeve, but I don't like things dangling like that to the right and the codebase is pretty consistent in that regard. This would be better:

```scala
    def evalCursor(
      ref: F[Iterant[F, A]], cursor: BatchCursor[A], rest: F[Iterant[F, A]],
      stop: F[Unit], dropped: Int, index: Int, continueDropping: Boolean): Iterant[F, A] = { 

      // ....
```